File: algorithms-data-structures/ts/src/algorithms/mixed/two-sum-ii/sum.ts
export function twoSum(numbers: number[], target: number): number[] {
  let left = 0;
  let right = numbers.length - 1;
  while (left <= right) {
    let tempTarget = numbers[left] + numbers[right];
    if (tempTarget === target) return [left + 1, right + 1];
    if (tempTarget < target) {
      left++;
    } else {
      right--;
    }
  }
  return [];
}

File: go/src/algorithms/mixed/two-sum-ii/sum.go
package twosumii

func twoSum(numbers []int, target int) []int {
	left, right := 0, len(numbers)-1
	for left <= right {
		tempTarget := numbers[left] + numbers[right]
		if tempTarget == target {
			return []int{left + 1, right + 1}
		} else if tempTarget < target {
			left++
		} else if tempTarget > target {
			right--
		}
	}
	return []int{0, 0}
}
